internal/credential/credential.go

```diff
@@ -119,6 +121,11 @@ type Password string
 // PrivateKey represents a secret private key.
 type PrivateKey []byte
 
+// JsonObject represents a JSON object that is serialized.
+type JsonObject struct {
```

Comment:
It seems like some of the code later could be slightly simplified by doing `type JsonObject structpb.Struct` instead of embedding but I'm not sure it makes any true difference.

Reply:
I decided to use struct composition instead so I can use some of the methods from structpb.Struct. For example, I use the AsMap() function.